src/Types/RelationParserHelper.php

```diff
@@ -47,14 +69,27 @@ public function findRelatedModelInRelationMethod(string $fileName, string $metho
             return null;
         }
 
-        $argValue = $methodCall->args[0]->value;
+        $scope = $this->scopeFactory->create(
+            ScopeContext::create($fileName),
+            false,
+            $methodReflection
+        );
+
+        $methodScope = $scope
+            ->enterClass($methodReflection->getDeclaringClass())
+            ->enterClassMethod($relationMethod, TemplateTypeMap::createEmpty(), [], null, null, null, false, false, false);
 
-        if ($argValue instanceof Node\Expr\ClassConstFetch && $argValue->class instanceof Node\Name) {
-            return $argValue->class->toString();
+        $argType = $methodScope->getType($methodCall->args[0]->value);
+
+        if ($argType instanceof ConstantStringType) {
+            return $argType->getValue();
         }
 
-        if ($argValue instanceof Node\Scalar\String_) {
-            return $argValue->value;
+        if ($argType instanceof GenericClassStringType) {
+            /** @var ObjectType $modelType */
+            $modelType = $argType->getGenericType();
```

Comment:
This doesn't always have to be `ObjectType`. `@template T` without bound is TemplateMixedType and it doesn't have `getClassName()` on it. So you should add another `if`.